src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java

```diff
@@ -413,10 +431,7 @@ public void setApiUri(@CheckForNull String apiUri) {
             return;
```

Comment:
No, we actually don't want to throw an exception here.  
One the things that is most striking about this PR is how much of it goes against what we all agree is "good design" because it is serving the needs of data binding.  This is another one of those cases.  If this gets called more than once due to databinding we don't want to throw.

Reply:
In what scenarios are the setters called more than once? That seems surprising to me, is it related to `bindResolve`? Otherwise, I would think that we definitely would want to throw exceptions as @darxriggs describes, otherwise, something like this ordering of events (only possible via incorrect data binding): `setRepoOwner`, followed by `setRepositoryUrl`, followed by `setRepository`, would silently leave the `GitHubSCMSource` in an invalid state. Is there a guaranteed order in which setters will be called?